File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
#  20분정도 소요
def solution(sizes):
    answer = 1
    a, b = [], []
    min_num = 0
    max_num = 0
    for i, j in sizes:
        if (min(i,j) > min_num):
            min_num = min(i,j)
        if (max(i,j) > max_num):
            max_num = max(i,j)
            
    # print(max_num, min_num)
    return max_num * min_num

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
// 문제풀이 시간: 1st - 1시간 반 소요 후 실패, 2nd - 질문하기 힌트(https://school.programmers.co.kr/questions/20800 )보고 20분 소요 후 성공

import java.util.*;
class Solution {
    public int solution(int[][] sizes) {
        int max_w = 0;
        int max_h = 0;
        
        for (int i=0; i<sizes.length; i++) {
            max_w = Math.max(max_w, Math.max(sizes[i][0], sizes[i][1]));
            max_h = Math.max(max_h, Math.min(sizes[i][0], sizes[i][1]));
        }
        return max_w * max_h;
    }
}
